pkg/scheduler/partition.go

```diff
@@ -425,6 +425,10 @@ func (pc *PartitionContext) removeAppInternal(appID string) *objects.Application
 	return app
 }
 
+func (pc *PartitionContext) GetApplication(appID string) *objects.Application {
```

Comment:
hi @chia7712 thanks for review.
The are so many places call the internal method.
Which way do you think is better?
First,
- Add RLock/RUnlock in exposed method & Remove RLock/RUnlock in internal method
- Add explicit RLock/RUnlock snippet in every place that call the internal method
Second,
Just change the exposed name

Reply:
> Add RLock/RUnlock in exposed method & Remove RLock/RUnlock in internal method

yep

> Add explicit RLock/RUnlock snippet in every place that call the internal method

How about just replacing internal method `getApplication` by public version `GetApplication`? Small change is good at avoiding new bugs :)

Also, you can file new Jira as follow-up to improve lock usage if you observe any weird use case.

Reply:
It might be a package internal method until this point. Lots of places calling it but they are all in the package. The partition is the place where the events from the shim get processed **and** the scheduling cycle runs from. That means we have multiple go routines that could be changing values stored in the partition. The locking makes sure we have no data races and a consistent view.
The internal method was locked for that reason and needs to stay like that.